### ISSUE TITLE
False warnings from clang's scan-build

### ISSUE THREAD

**Issue by gperciva** [Tarsnap/spiped#330]
clang 13.0.0's scan-build reports:

* Logic error "Result of operation is garbage or undefined" in `crypto_aesctr_stream_cipherblock_use`.

  This is discussed in https://github.com/Tarsnap/libcperciva/issues/372.

  TL;DR: clang doesn't realize that writing a 32-bit integer will initialize all 4 bytes.

* Logic error "Assigned value is garbage or undefined" in `_mm_loadu_si64` (called from `cpusupport-X86-AESNI.c`)

  This is discussed in https://github.com/Tarsnap/libcperciva/issues/423.
  
  TL;DR: clang-scan doesn't like its own `<emmintrin.h>` header.
